pkg/frontend/transport/handler.go

```diff
@@ -269,7 +271,12 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		writeServiceTimingHeader(queryResponseTime, hs, stats)
 	}
 
-	logger := util_log.WithContext(r.Context(), f.log)
+	userAgent := r.Header.Get("User-Agent")
+	if userAgent == "" {
+		userAgent = userAgentUnKnown
+	}
+
+	logger := util_log.WithUserAgent(userAgent, util_log.WithContext(r.Context(), f.log))
```

Comment:
Why do we need this log wrapper? This seems too much to me.
We already log user agent. Usually the practice is `util_log.WithContext` will log your request ID and you should correlate user agent with your request ID, instead of always logging user agent

Reply:
I see. How about keeping existing?

Reply:
Thanks, I fixed it.